authors/apps/profiles/views.py

```diff
@@ -122,24 +119,41 @@ def patch(self, request, username):
                 }
             })
         try:
-            avatar_data = str(request.data.get('avatar'))
+            upload_info = request.data.get('avatar')
+            content = request.FILES.get('avatar')
+            if not content:
+                APIException.status_code = status.HTTP_400_BAD_REQUEST
+                raise APIException(
+                    {
+                        "message": "You have not supplied any file for upload"
+                    })
+            size = content.size
+            avatar_data = str(upload_info)
             if not avatar_data.lower().endswith((".png", ".jpg", ".jpeg")):
                 # checks extension of image complies with 'png','jpg' and 'jpeg' formats
+                APIException.status_code = status.HTTP_400_BAD_REQUEST
                 raise APIException(
-                    {"message": "This is an invalid avatar format"})
-            data = {"avatar": request.data.get('avatar')}
+                    {
+                        "message": "This is an invalid avatar format, '.jpg','.jpeg', '.png' are the accepted formats"
+                    })
+
+            if size > 5021440:
+                # checks that avatar size does not exceed 5mb
+                APIException.status_code = status.HTTP_400_BAD_REQUEST
+                raise APIException(
+                    {"message": "This is image is too large avatars cannot be more than 5mb"})
+
```

Comment:
In the response message, you could add a comma between 'large' and 'avatars' for the message to more coherent

Reply:
Let me look into that.

Reply:
@ipaullly Kindly confirm that the changes have been effected accordingly.